.gitlab-ci.yml

```diff
@@ -0,0 +1,125 @@
+        docker tag "$IMG_TAG" "$BASE_PUSH_TARGET:$BUILD_STAMP"
+        docker push "$BASE_PUSH_TARGET:$BUILD_STAMP"
+                
+        docker tag "$IMG_TAG" "$BASE_PUSH_TARGET:$BUILD_VERSION"
+        docker push "$BASE_PUSH_TARGET:$BUILD_VERSION"
+        
+        docker tag "$IMG_TAG" "$BUILD_IMG_TAG"
+        docker push "$BUILD_IMG_TAG" 
+      fi
+Test-Image:
+  tags: 
+    - docker-executor
+
+  stage: Test
+  image: ${DOCKER_TESTER}
+  before_script:
+    - source shared-vars.sh
+  
+  script:
+    - yum install openssh openssh-clients -y 
+    - |
+      if [[ ! ( -z ${BASTION_KEY} ) && ! ( -z ${BASTION}) ]]; then
+        set -x
+        SESSION=$(echo $BUILD_IMG_TAG | awk -F ':' '{print $2}')
+        chmod 600 ${BASTION_KEY}
+        ssh -o "StrictHostKeyChecking=no" -i ${BASTION_KEY} -t ${BASTION} "sed -i  '/bigip_controller_img/c\    \"bigip_controller_img\": \"${BUILD_IMG_TAG}\",' ~/testenv_symbols/testenv_symbols.json"
+        if [[ ${ORCHESTRATION} == "k8s" ]] ; then
+          ssh -o "StrictHostKeyChecking=no" -i  ${BASTION_KEY} -t ${BASTION} "cd ${TEST_DIR} && scripts/run_systests.sh  --project=bigip_ctlr --session=${SESSION}-cluster --include-tags=${INCLUDE_TAGS} --orchestration=k8s --pool-mode=cluster --namespace-mode=${NAMESPACE_MODE} --exclude-tags=${EXCLUDE_TAGS}" ;
+          ssh -o "StrictHostKeyChecking=no" -i  ${BASTION_KEY} -t ${BASTION} "cd ${TEST_DIR} && scripts/run_systests.sh  --project=bigip_ctlr --session=${SESSION}-nodeport --include-tags=${INCLUDE_TAGS} --orchestration=k8s --pool-mode=nodeport --namespace-mode=${NAMESPACE_MODE} --exclude-tags=${EXCLUDE_TAGS}" ;
+          
+        elif [[ ${ORCHESTRATION} == "oscp" || ${ORCHESTRATION} == "os" ]] ; then
+          ssh -o "StrictHostKeyChecking=no" -i  ${BASTION_KEY} -t ${BASTION} "cd ${TEST_DIR} && scripts/run_systests.sh   --project=bigip_ctlr --session=${SESSION}-cluster --include-tags=${INCLUDE_TAGS} --orchestration=openshift --pool-mode=cluster --namespace-mode=${NAMESPACE_MODE} --exclude-tags=${EXCLUDE_TAGS}" ;
+          ssh -o "StrictHostKeyChecking=no" -i  ${BASTION_KEY} -t ${BASTION} "cd ${TEST_DIR} && scripts/run_systests.sh   --project=bigip_ctlr --session=${SESSION}-nodeport --include-tags=${INCLUDE_TAGS} --orchestration=openshift --pool-mode=nodeport --namespace-mode=${NAMESPACE_MODE} --exclude-tags=${EXCLUDE_TAGS}" ;
+          
+        else
+          echo "Invalid Orchestration. Exiting...";
+          exit 1;
+        fi ;
+      else
+        yum install -y epel-release && yum clean all
+        yum install git unzip wget make openssh python python-pip  -y
+        pip install ansible==2.6.5
+        # Cloning Glider Repo
+        export GIT_ASKPASS=${PWD}/git_config.sh
+        mkdir glider
+        echo $GLIDER_REPO
+        pushd glider
+        echo -e '#!/bin/bash \necho $GIT_PASSWORD' > $GIT_ASKPASS
+        chmod +x $GIT_ASKPASS
+        git init && git config http.sslVerify false
```

Comment:
Let's either use proper CA certificates to trust or use SSH Keys for authentication.